File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TDBXInt16Value.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXInt16Value.h"


@implementation TDBXInt16Value
-(id)init{
	self = [super init];
	if (self) {
		[self setDBXType:Int16Type];
	}
	return self;
}
-(void) dealloc{
	[super dealloc];
}
-(void) SetNull {
	ValueNull = YES;
	DBXInternalValue = 0;
}
-(bool) isNull {
	return ValueNull;
}
-(void) SetAsInt16:(int)value{
	DBXInternalValue = value;
}
-(int) GetAsInt16{
	return DBXInternalValue;
}
@end

File: Demos/Datasnap/Server/proxy/java_android/com/embarcadero/javaandroid/TDBXInt16Value.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javaandroid;

/**
 * 
 * Wraps the Int16 type and allows it to be null
 *
 */

public class TDBXInt16Value extends DBXValue {
	protected boolean ValueNull = false;
	private int DBXInt16Value;	
	
	public TDBXInt16Value() {
		super();
		setDBXType(DBXDataTypes.Int16Type);
	}
	
	
	@Override
	public void setNull() {
		ValueNull = true;
		DBXInt16Value = 0;
	}

	public boolean isNull() {
		return ValueNull;
	}
	
	@Override
	public void SetAsInt16(int Value) throws DBXException {
		ValueNull = false;
		DBXInt16Value = Value;
	}
	
	@Override
	public int GetAsInt16() throws DBXException {
		return DBXInt16Value;
	}	

}
